integration-tests/scaffolding/helpers.ts

```diff
@@ -1,22 +1,166 @@
+import { createKeys } from "./apiConnection";
+import { getAccountInfo, transferFunds } from "./extrinsicHelpers";
 
-export type EventMap = {[key: string]: Event}
-export type Sr25519Signature = {Sr25519: `0x${string}`}
+export const INITIAL_FUNDING = 100000000n;
 
-export function groupEventsByKey() {
+export enum DevAccounts {
+    Alice = "//Alice",
+    Bob = "//Bob",
+    Charlie = "//Charlie",
+    Dave = "//Dave",
+    Eve = "//Eve",
+    Ferdie = "//Ferdie",
+}
+
+export interface AccountFundingInputs {
+    api: ApiRx;
+    amount: bigint;
+    source: string;
+    name?: string;
+    context?: string;
+}
+
+export const generateFundingInputs = (api: ApiRx, context?: string) => ({
+    api,
+    amount: INITIAL_FUNDING,
+    source: DevAccounts.Alice,
+    context
+});
+
+export class EventError extends Error {
+    name: string = '';
+    message: string = '';
+    stack?: string = '';
+    section?: string = '';
+    rawError: DispatchError;
+
+    constructor(source: DispatchError) {
+        super();
+
+        if (source) {
+            if (source.isModule) {
+                const decoded = source.registry.findMetaError(source.asModule);
+                this.name = decoded.name;
+                this.message = decoded.docs.join(' ');
+                this.section = decoded.section;
+            } else {
+                this.name = source.type;
+                this.message = source.type;
+                this.section = '';
+            }
+        }
+        this.rawError = source;
+    }
+
+    public toString() {
+        return `${this.section}.${this.name}: ${this.message}`;
+    }
+}
+
+export type EventMap = { [key: string]: Event }
+export type Sr25519Signature = { Sr25519: `0x${string}` }
+
+interface BeginEndBalances {
+    beginBalance: bigint;
+    endBalance?: bigint;
+}
+let balancesMap = new Map<string, Map<string, BeginEndBalances>>;
+let doTxAccounting = false;
+export let txAccountingHook = async (api, context?): Promise<void> => { };
+
+if (process.env.ENABLE_TX_ACCOUNTING === 'true' || process.env.ENABLE_TX_ACCOUNTING === '1') {
+    doTxAccounting = true;
+    txAccountingHook = async (api, context?) => showTotalCost(api, context);
+}
+
+export function parseResult() {
     return pipe(
-        map((result: ISubmittableResult) => result.events.reduce((acc, {event}) => { acc[eventKey(event)] = event.data; return acc}, {})),
-    )
+        tap((result: ISubmittableResult) => {
+            if (result.dispatchError) {
+                let err: EventError;
+
+                switch (result.dispatchError.type) {
+                    case 'Module': {
+                        err = new EventError(result.dispatchError);
+                        break;
+                    }
+
+                    default: {
+                        err = new EventError(result.dispatchError);
+                        break;
+                    }
+                }
+
+                console.log(err.toString());
```

Comment:
Do we want this console.log?

Reply:
Couldn't decide.
More cleanup & README to come in follow-on ticket. Or I can push & get re-review/re-approval. I've already got quite a bit queued up for followup, though, and I'd like to get this in because it enables the possibility of someone starting to integrate this into the CI pipeline.